libtransmission/utils.c

```diff
@@ -587,6 +587,35 @@ char* tr_strsep(char** str, char const* delims)
 #endif
 }
 
+char* tr_strjoin(char** arr, size_t len, char const* delim)
```

Comment:
Since we're not changing the array, consider
```suggestion
char* tr_strjoin(char const* const* arr, size_t len, char const* delim)
```